mp/src/game/shared/momentum/run/mom_run_safeguard.cpp

```diff
@@ -0,0 +1,96 @@
+#include "cbase.h"
+
+#include "mom_run_safeguard.h"
+#include "in_buttons.h"
+#include "mom_player_shared.h"
+
+#ifdef CLIENT_DLL
+#include "input.h"
+#endif
+
+#include "tier0/memdbgon.h"
+
+#define MOM_DOUBLEPRESS_MAXTIME 0.5f
+
+enum MomRunSafeguard_t
+{
+    MOM_RUN_SAFEGUARD_NONE = 0,
+    MOM_RUN_SAFEGUARD_MOVEMENTKEYS,
+    MOM_RUN_SAFEGUARD_DOUBLEPRESS,
+
+    MOM_RUN_SAFEGUARD_FIRST = MOM_RUN_SAFEGUARD_NONE,
+    MOM_RUN_SAFEGUARD_LAST = MOM_RUN_SAFEGUARD_DOUBLEPRESS
+};
+
+static MAKE_CONVAR(mom_run_safeguard, "1", FCVAR_ARCHIVE | FCVAR_REPLICATED,
+    "Changes the safeguard for run interfering operations (enabling practice mode, teleporting to saveloc, etc.).\n"
+    "0 = OFF,\n"
+    "1 = Enable when not pressing any movement keys,\n"
+    "2 - Enable on double press",
+    MOM_RUN_SAFEGUARD_FIRST, MOM_RUN_SAFEGUARD_LAST);
+
+MomRunSafeguard::MomRunSafeguard(const char *szAction) : m_flLastTimePressed(0.0f), m_bDoublePressSafeguard(true)
+{
+    Q_strncpy(m_szAction, szAction, BUFSIZELOCL);
+}
+
+bool MomRunSafeguard::IsSafeguarded()
+{
+#ifdef CLIENT_DLL
+    const auto pPlayer = C_MomentumPlayer::GetLocalMomPlayer();
+    if (!pPlayer)
+        return false; // no safeguard needed if not local player
+
+    const auto pEntData = pPlayer->GetCurrentUIEntData();
+    if (!pEntData)
+        return false;
+
+    if (!pEntData->m_bTimerRunning || pPlayer->GetCurrentUIEntity()->GetEntType() >= RUN_ENT_GHOST)
+        return false; // no safeguard if timer is off or not the local player 
```

Comment:
You already get the local player, so you should just get the ent data from them, and if you don't want spectating stuff you can check to see if they aren't an observer.